compiler/luci/export/src/CircleExporterImpl.cpp

```diff
@@ -222,19 +222,42 @@ void CircleExporterImpl::exportModule(Module *module)
   std::string description_str = "nnpackage";
   auto description = _builder.CreateString(description_str);
 
-  // create array of buffers
-  auto buffers = _builder.CreateVector(md._buffers);
-
-  // empty metadata
-  std::vector<int> metadata_buffer_vec;
-  auto metadata_buffer = _builder.CreateVector(metadata_buffer_vec);
+  // metadata
+  std::vector<flatbuffers::Offset<circle::Metadata>> metadata_vec;
+  {
+    // Insert origin node names
+    auto origin_name_buffer_id = static_cast<uint32_t>(md._buffers.size());
+    auto origin_name_data = std::vector<uint8_t>(); // TODO Fill data using md._metadata
+    auto origin_name_vec_offset =
+      _builder.CreateVector(origin_name_data.data(), origin_name_data.size());
+    auto origin_name_offset = CreateBuffer(_builder, origin_name_vec_offset);
+    md._buffers.push_back(origin_name_offset);
+    Offset<circle::Metadata> origin_name_metadata =
+      CreateMetadata(_builder, _builder.CreateString("origin_names_list"), origin_name_buffer_id);
+    metadata_vec.push_back(origin_name_metadata);
+
+    // Insert origin mapping data
+    auto origin_map_buffer_id = static_cast<uint32_t>(md._buffers.size());
+    auto origin_map_data = std::vector<uint8_t>(); // TODO Fill data using md._metadata
+    auto origin_map_vec_offset =
+      _builder.CreateVector(origin_map_data.data(), origin_map_data.size());
+    auto origin_map_offset = CreateBuffer(_builder, origin_map_vec_offset);
+    md._buffers.push_back(origin_map_offset);
+    Offset<circle::Metadata> origin_map_metadata =
+      CreateMetadata(_builder, _builder.CreateString("origin_mapping"), origin_map_buffer_id);
+    metadata_vec.push_back(origin_map_metadata);
+  }
+  auto metadata = _builder.CreateVector(std::vector<Offset<Metadata>>(metadata_vec));
 
   // This version is taken from comment in fbs
   constexpr uint32_t version = 0;
 
+  // create array of buffers
+  auto buffers = _builder.CreateVector(md._buffers);
+
   // Model
   auto model_offset = CreateModel(_builder, version, operator_codes, subgraphs, description,
-                                  buffers, metadata_buffer);
+                                  buffers, 0 /* metadata_buffer */, metadata);
```

Comment:
You set `metadata_buffer ` to 0 and added `metadata`. Could you explain about this? 
I'm asking because I'm not good at this part...

Reply:
@seanshpark 
```flatbuffers
...
// Deprecated, prefer to use metadata field.
  metadata_buffer:[int];
```
`metadata_buffer` seems deprecated. This is why he did the way he does.